diff --git a/emar.py b/emar.py
index 1af065aa1..2b03753ce 100755
--- a/emar.py
+++ b/emar.py
@@ -67,6 +67,10 @@ def run():
         full_name = os.path.abspath(orig_name)
         dirname = os.path.dirname(full_name)
         basename = os.path.basename(full_name)
+        # https://github.com/emscripten-core/emscripten/issues/9733
+        if basename not in contents:
+          contents.add(basename)
+          continue
 
         h = hashlib.md5(full_name.encode('utf-8')).hexdigest()[:8]
         parts = basename.split('.')
diff --git a/emcc.py b/emcc.py
index ccb06e3bf..61294510f 100755
--- a/emcc.py
+++ b/emcc.py
@@ -1884,6 +1884,10 @@ There is NO warranty; not even for MERCHANTABILITY or FITNESS FOR A PARTICULAR P
         return any(flag.startswith(x) for x in ('-l', '-L', '-Wl,'))
 
       compile_args = [a for a in newargs if a and not is_link_flag(a)]
+      # https://github.com/emscripten-core/emscripten/issues/9013
+      if '-fPIC' in compile_args and not shared.Settings.RELOCATABLE:
+        shared.warning('Ignoring -fPIC flag when not building with SIDE_MODULE or MAIN_MODULE')
+        compile_args.remove('-fPIC')
 
       # Bitcode args generation code
       def get_clang_command(input_files):
diff --git a/src/library_html5.js b/src/library_html5.js
index 267dbe30e..d0ffda384 100644
--- a/src/library_html5.js
+++ b/src/library_html5.js
@@ -1533,7 +1533,9 @@ var LibraryJSEvents = {
       return {{{ cDefine('EMSCRIPTEN_RESULT_NOT_SUPPORTED') }}};
     }
 
-    if (__currentFullscreenStrategy.canvasResizedCallback) {
+    // https://github.com/emscripten-ports/SDL2/issues/87
+    //if (__currentFullscreenStrategy.canvasResizedCallback) {
+    if (false) {
 #if USE_PTHREADS
         if (__currentFullscreenStrategy.canvasResizedCallbackTargetThread) JSEvents.queueEventHandlerOnThread_iiii(__currentFullscreenStrategy.canvasResizedCallbackTargetThread, __currentFullscreenStrategy.canvasResizedCallback, {{{ cDefine('EMSCRIPTEN_EVENT_CANVASRESIZED') }}}, 0, __currentFullscreenStrategy.canvasResizedCallbackUserData);
         else
