diff --git a/tools/system_libs.py b/tools/system_libs.py
index 11fe8646f..02a772513 100755
--- a/tools/system_libs.py
+++ b/tools/system_libs.py
@@ -53,7 +53,8 @@ def dir_is_newer(dir_a, dir_b):
   assert os.path.exists(dir_b)
   newest_a = max([os.path.getmtime(x) for x in get_all_files_under(dir_a)])
   newest_b = max([os.path.getmtime(x) for x in get_all_files_under(dir_b)])
-  return newest_a < newest_b
+  # https://github.com/emscripten-core/emscripten/issues/9342
+  return newest_a > newest_b
 
 
 def get_cflags(force_object_files=False):
--- emscripten-git/emcc.py	2020-09-10 12:56:16.876047227 +0200
+++ upstream/emscripten/emcc.py	2020-09-10 12:57:09.611839316 +0200
@@ -1305,7 +1305,8 @@
       else:
         link_to_object = True
 
-    if not link_to_object and not compile_only and final_suffix not in EXECUTABLE_ENDINGS:
+    # https://github.com/emscripten-core/emscripten/issues/12148
+    if not link_to_object and not compile_only and final_suffix not in EXECUTABLE_ENDINGS and not os.environ.get('EMMAKEN_JUST_CONFIGURE'):
       # TODO(sbc): Remove this emscripten-specific special case.  We should only generate object
       # file output with an explicit `-c` or `-r`.
       diagnostics.warning('emcc', 'assuming object file output, based on output filename alone.  Add an explict `-c`, `-r` or `-shared` to avoid this warning')
--- emscripten/src/library_idbfs.js	2020-09-02 17:52:17.000000000 +0200
+++ emscripten/src/library_idbfs.js	2020-10-02 16:18:33.566465230 +0200
@@ -231,7 +231,8 @@
       Object.keys(src.entries).forEach(function (key) {
         var e = src.entries[key];
         var e2 = dst.entries[key];
-        if (!e2 || e['timestamp'] > e2['timestamp']) {
+        // https://github.com/emscripten-core/emscripten/issues/12348
+        if (!e2 || e['timestamp'] != e2['timestamp']) {
           create.push(key);
           total++;
         }
--- emscripten/system/bin/sdl2-config	2020-09-02 17:52:17.000000000 +0200
+++ emscripten/system/bin/sdl2-config	2020-10-16 17:07:51.734317365 +0200
@@ -10,5 +10,6 @@
 if '--cflags' in args or '--libs' in args:
   print('-s USE_SDL=2')
 elif '--version' in args:
-  print('2.0.0')
+  # https://github.com/emscripten-core/emscripten/pull/12538
+  print('2.0.10')
 
