diff --git a/tools/system_libs.py b/tools/system_libs.py
index 11fe8646f..02a772513 100755
--- a/tools/system_libs.py
+++ b/tools/system_libs.py
@@ -63,7 +63,8 @@ def dir_is_newer(dir_a, dir_b):
   assert os.path.exists(dir_b)
   newest_a = max([os.path.getmtime(x) for x in get_all_files_under(dir_a)])
   newest_b = max([os.path.getmtime(x) for x in get_all_files_under(dir_b)])
-  return newest_a < newest_b
+  # https://github.com/emscripten-core/emscripten/issues/9342
+  return newest_a > newest_b
 
 
 def get_cflags(force_object_files=False):
diff --git a/src/library_html5.js b/src/library_html5.js
index 0b6e3cd78c3..ef3040b8619 100644
--- a/src/library_html5.js
+++ b/src/library_html5.js
@@ -39,13 +39,16 @@ var LibraryJSEvents = {
     removeEventListenersRegistered: false, 
 
 #if HTML5_SUPPORT_DEFERRING_USER_SENSITIVE_REQUESTS
-    // If positive, we are currently executing in a JS event handler.
-    inEventHandler: 0,
     // If we are in an event handler, specifies the event handler object from the eventHandlers array that is currently running.
     currentEventHandler: null,
 #endif
 */
 
+    // https://github.com/emscripten-core/emscripten/issues/11515
+    // If positive, we are currently executing in a JS event handler.
+    // (this particular property must be initialized to zero, as we ++/-- it)
+    inEventHandler: 0,
+
     removeAllEventListeners: function() {
       for(var i = JSEvents.eventHandlers.length-1; i >= 0; --i) {
         JSEvents._removeHandler(i);
