https://github.com/emscripten-core/emscripten/issues/7908

--- a/src/video/emscripten/SDL_emscriptenopengles.c     2018-03-01 17:34:42.000000000 +0100
+++ b/src/video/emscripten/SDL_emscriptenopengles.c     2019-01-22 16:57:56.270005327 +0100
@@ -87,7 +87,7 @@
     ES2 renderer resets state on init anyway, clearing the canvas should be enough
     */
 
-    glClear(GL_COLOR_BUFFER_BIT | GL_DEPTH_BUFFER_BIT | GL_STENCIL_BUFFER_BIT);
+    //glClear(GL_COLOR_BUFFER_BIT | GL_DEPTH_BUFFER_BIT | GL_STENCIL_BUFFER_BIT);
 
     SDL_EGL_DeleteContext(_this, context);
 }
