Description: Support opening new browser tabs from Python/Emscripten
Forwarded: no
Author: Sylvain Beucler <beuc@beuc.net>
Last-Update: 2018-10-28

--- Python-2.7.10/Lib/webbrowser.py	2018-10-19 21:48:22.701879860 +0200
+++ Python-2.7.10/Lib/webbrowser.py	2018-10-19 21:53:22.680671785 +0200
@@ -6,7 +6,7 @@
 import shlex
 import sys
 import stat
-import subprocess
+#import subprocess
 import time
 
 __all__ = ["Error", "open", "open_new", "open_new_tab", "get", "register"]
@@ -656,6 +656,16 @@
              GenericBrowser(["start", "netscape", "%s"]), -1)
 
 
+class EmscriptenBrowser(BaseBrowser):
+    def open(self, url, new=0, autoraise=True):
+        import emscripten
+        emscripten.run_script("""try { window.open('%s', '_blank'); } catch (e) { console.log(e); }""" % url.replace("'", "%27"))
+        return True
+
+if sys.platform == 'emscripten':
+    register("emscripten", None, EmscriptenBrowser())
+
+
 # OK, now that we know what the default preference orders for each
 # platform are, allow user to override them with the BROWSER variable.
 if "BROWSER" in os.environ:
